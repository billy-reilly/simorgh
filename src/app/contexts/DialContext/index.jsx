import React from 'react';
import { any, objectOf, node } from 'prop-types';

export const DialContext = React.createContext({});

export const DialContextProvider = ({ dials, children }) => (
  <DialContext.Provider value={dials}>{children}</DialContext.Provider>
);

DialContextProvider.propTypes = {
  dials: objectOf(any).isRequired,
  children: node.isRequired,
};
