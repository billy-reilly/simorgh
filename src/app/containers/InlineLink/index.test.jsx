import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import InlineLinkContainer from './index';
import { ServiceContextProvider } from '../../contexts/ServiceContext';

const fragmentBlock = (text, attributes = []) => ({
  type: 'fragment',
  model: {
    text,
    attributes,
  },
});

const testInternalInlineLink = (description, locator, blocks, isExternal) => {
  shouldMatchSnapshot(
    description,
    /*
      for the value it would bring, it is much simpler to wrap a react-router Link in a Router, rather than mock a Router or pass some mocked context.
    */
    <StaticRouter>
      <InlineLinkContainer
        locator={locator}
        blocks={blocks}
        isExternal={isExternal}
      />
    </StaticRouter>,
  );
};

describe('InlineLinkContainer', () => {
  describe('link matching routes for SPA', () => {
    testInternalInlineLink(
      'should render correctly',
      'https://www.bbc.com/news/articles/c0g992jmmkko',
      [fragmentBlock('This is text for an internal link')],
      false,
    );

    testInternalInlineLink(
      'should render correctly for TEST environment',
      'https://www.test.bbc.com/news/articles/c0g992jmmkko',
      [fragmentBlock('This is text for an internal link')],
      false,
    );
  });

  describe('internal link not matching SPA route', () => {
    shouldMatchSnapshot(
      'should render correctly',
      <InlineLinkContainer
        locator="https://www.bbc.com/news"
        blocks={[fragmentBlock('This is bold text for a link', ['bold'])]}
        isExternal={false}
      />,
    );
  });

  describe('external link accessibility', () => {
    shouldMatchSnapshot(
      'should be explicitly marked "external" for screen reader users',
      <ServiceContextProvider service="news">
        <InlineLinkContainer
          locator="https://www.example.com/"
          blocks={[fragmentBlock('This is a link', [])]}
          isExternal
        />
      </ServiceContextProvider>,
    );

    shouldMatchSnapshot(
      'should be explicitly marked "external" for screen reader users & localised',
      <ServiceContextProvider service="persian">
        <InlineLinkContainer
          locator="https://www.example.com/"
          blocks={[fragmentBlock('این لینک هست', [''])]}
          isExternal
        />
      </ServiceContextProvider>,
    );
  });
});
