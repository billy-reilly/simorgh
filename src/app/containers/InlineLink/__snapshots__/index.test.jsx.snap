// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users & localised 1`] = `
.c0 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #757575;
  border-bottom: 1px solid #757575;
}

.c0:focus {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c0:hover {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c1 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

<a
  className="c0"
  href="https://www.example.com/"
>
  این لینک هست
  <span
    className="c1"
  >
     ،لینک خارجی
  </span>
</a>
`;

exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users 1`] = `
.c0 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #757575;
  border-bottom: 1px solid #757575;
}

.c0:focus {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c0:hover {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c1 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

<a
  className="c0"
  href="https://www.example.com/"
>
  This is a link
  <span
    className="c1"
  >
    , external
  </span>
</a>
`;

exports[`InlineLinkContainer internal link not matching SPA route should render correctly 1`] = `
.c0 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #757575;
  border-bottom: 1px solid #757575;
}

.c0:focus {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c0:hover {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

<a
  className="c0"
  href="https://www.bbc.com/news"
>
  <b>
    This is bold text for a link
  </b>
</a>
`;

exports[`InlineLinkContainer link matching routes for SPA should render correctly 1`] = `
.c0 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #757575;
  border-bottom: 1px solid #757575;
}

.c0:focus {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c0:hover {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

<a
  className="-Link c0"
  href="/news/articles/c0g992jmmkko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;

exports[`InlineLinkContainer link matching routes for SPA should render correctly for TEST environment 1`] = `
.c0 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #757575;
  border-bottom: 1px solid #757575;
}

.c0:focus {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

.c0:hover {
  color: #B80000;
  border-bottom: 2px solid #B80000;
}

<a
  className="-Link c0"
  href="/news/articles/c0g992jmmkko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;
